src/ai/defaultai.cc

```diff
@@ -748,8 +735,33 @@ void DefaultAI::late_initialization() {
 			}
 
 			// If this is a producer, does it act also as supporter?
-			if (!bo.ware_outputs.empty() && !bo.production_hints.empty()) {
+			if (!bo.ware_outputs.empty() && !prod.supported_productionsites().empty()) {
+				log_dbg_time(
+				   gametime, "AI %d detected supporting producer: %s", player_number(), bo.name);
+				for (const auto& supp : prod.supported_productionsites()) {
+					log_dbg_time(gametime, "  -> %s", supp.c_str());
+				}
 				bo.set_is(BuildingAttribute::kSupportingProducer);
+				/* Buildings detected at the time of writing:
+				 *
+				 *   frisians_clay_pit  -> frisians_aqua_farm, frisians_charcoal_burners_house
+				 *   frisians_farm      -> frisians_beekeepers_house
+				 *   frisians_reed_farm -> frisians_beekeepers_house
```

Comment:
frisians berry farm is missing

Reply:
If it is, this is already a bug in master. Here is a log from master for the Frisians:

```
[00:00:26.749 real] DEBUG: NOCOM supporting producer: frisians_clay_pit
[00:00:26.749 real] DEBUG: NOCOM supporting producer: frisians_farm
```

The reed farm is added in this branch, which is a change but correct, because it does support the beekeeper.

Reply:
i was wrong as berry farm is only a supporter and not a supporting producer